### PR TITLE
Documentation of ts_gtrends_windows

### LINKED ISSUE #3: Error (ts_gtrends_windows.R#23): `false` must be a `Date` object, not a `Date` object
ts_gtrends_windows produces the following error when called as

`ts_gtrends_windows('zara')`

>Error (ts_gtrends_windows.R#23): `false` must be a `Date` object, not a `Date` object

https://github.com/trendecon/trendecon/blob/8da50c5dcf50d9eba9e322c6fc2ef7984610b0ea/R/ts_gtrends_windows.R#L33

### PR DESCRIPTION
**Pull request by uwethuemmel**
Though not an exported function, I had to understand what `ts_gtrends_windows` does (and thus also documented it).

I refactored preparation of the time windows into a separate function (in the same file) as that made it easier for me to understand the structure. I hope that doing things like that is ok with you.

Also fixes #3 using a workaround.

### CODE REVIEW

File: R/ts_gtrends_windows.R [hunk 1]
```diff
@@ -23,13 +35,29 @@ ts_gtrends_windows <- function(keyword = NA,
...
   ) %>%
-    rowwise() %>%
-    mutate(end_date = seq(start_date, length.out = 2, by = windowsize)[2]) %>%
-    ungroup() %>%
-    # truncate end date to Sys.Date() to avoid google errors
+  rowwise() %>%
+  mutate(end_date = seq(start_date, length.out = 2, by = windowsize)[2]) %>%
+  ungroup()
+
+  tbl <- tbl[1:nrow(tbl),] %>%
```
Comment (christophsax) [inline R/ts_gtrends_windows.R:60]:
```r
mutate(end_date = as.Date(end_date)) %>%
```

is a bit cleaner and you don't need the intermediate `tbl`.

File: R/ts_gtrends_windows.R [hunk 2]
```diff
@@ -40,18 +68,14 @@ ts_gtrends_windows <- function(keyword = NA,
     }) %>%
     # Make sure the last window is large enough to yield the desired frequency
     rowwise() %>%
-    mutate(start_date = ifelse(prevent_window_shrinkage & end_date == Sys.Date(), seq(start_date, length.out = 2, by = paste0("-", windowsize))[1], start_date)) %>%
+    mutate(start_date = ifelse(prevent_window_shrinkage & end_date == Sys.Date(),
```
Comment (christophsax) [inline R/ts_gtrends_windows.R:71]: just cosmetics, but can you do: https://style.tidyverse.org/syntax.html#long-lines

File: R/ts_gtrends_windows.R (hunk 1 above)
Comment (uwethuemmel) [reply R/ts_gtrends_windows.R:60]: unfortunately, this does not resolve the error, see here: https://github.com/trendecon/trendecon/issues/3#issuecomment-632089964

Comment (christophsax) [reply R/ts_gtrends_windows.R:78]: Sorry, this one:

    mutate(date = as.Date(as.numeric(date), origin = "1970-01-01")) %>%

File: R/ts_gtrends_windows.R (hunk 2 above)
Comment (uwethuemmel) [reply R/ts_gtrends_windows.R:88]: Good to know, thanks. Adjusted layout in 79874f6

File: R/ts_gtrends_windows.R (hunk 1 above)
Comment (uwethuemmel) [reply R/ts_gtrends_windows.R]: Indeed. Fixed here d7b9734

**Pull request merged**